### ISSUE TITLE
Executing scripts with `--package` dependencies only works the second time.

### ISSUE THREAD

**Issue by soenkehahn** [commercialhaskell/stack#444]
## Steps to reproduce

1. Create a file `Test.hs`:
``` haskell
import Safe

main :: IO ()
main = do
  print $ headMay ([] :: [Int])
```
2. Run this in a shell:
``` shell
$ rm -rf ~/.stack
$ stack --resolver lts-2.9 --install-ghc runghc --package safe Test.hs
Downloaded lts-2.9 build plan.
Caching build plan
Fetched package index.
Populated index cache.
safe-0.3.9: download
safe-0.3.9: configure
safe-0.3.9: build
safe-0.3.9: install

Test.hs:2:8:
    Could not find module ‘Safe’
    Use -v to see a list of the files searched for.
$ stack --resolver lts-2.9 --install-ghc runghc --package safe Test.hs
Nothing
$ stack --version
Version 0.1.1.0, Git revision fe18e81c5d82e876b6837506b4a8c1ab518caaf9
```

## Expected

It should work the first time.

**Comment by snoyberg** [comment]
Good catch. I've got a test case for this and a patch. Once I'm back home in an hour or so I'll push it.

**Comment by snoyberg** [comment]
Should be fixed by 513d136, can you confirm?

**Comment by soenkehahn** [comment]
Yes, works. Thanks for the quick fix.
(Closing.)
